pkg/s3select/progress.go

```diff
@@ -92,30 +93,35 @@ func (pr *progressReader) Stats() (bytesScanned, bytesProcessed int64) {
 }
 
 func newProgressReader(rc io.ReadCloser, compType CompressionType) (*progressReader, error) {
+	if rc == nil {
+		return nil, errors.New("newProgressReader: nil reader provided")
+	}
 	scannedReader := newCountUpReader(rc)
-	var r io.Reader
+	pr := progressReader{
+		rc:            rc,
+		scannedReader: scannedReader,
+	}
 	var err error
+	var r io.Reader
 
 	switch compType {
 	case noneType:
 		r = scannedReader
 	case gzipType:
-		r, err = gzip.NewReader(scannedReader)
+		pr.gzr, err = gzip.NewReader(scannedReader)
 		if err != nil {
 			if errors.Is(err, gzip.ErrHeader) || errors.Is(err, gzip.ErrChecksum) {
 				return nil, errInvalidGZIPCompressionFormat(err)
 			}
 			return nil, errTruncatedInput(err)
 		}
+		r = pr.gzr
 	case bzip2Type:
 		r = bzip2.NewReader(scannedReader)
```

Comment:
would it still be problem if it was bzip2 @klauspost ?

Reply:
@harshavardhana No, it doesn't require closing, it can be GCed as-is.

pgzip has readahead which keeps data unless it reaches the end of the stream, which is why the repro needs a `limit 1`.

Reply:
ah okay @klauspost